### ISSUE TITLE
Prevent the use of the `v-bind` shorthand `:` on directives

### ISSUE THREAD

**Issue by peteruithoven** [vuejs/eslint-plugin-vue#1277]
**Please describe what the rule should do:**

Prevent the use of the `v-bind` shorthand `:` on directives. 

**What category should the rule belong to?**

- [ ] Enforces code style (layout)
- [x] Warns about a potential error (problem)
- [x] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<input :v-model="message" >

<div :v-if="show" >Hello</div>

<input type="checkbox" 
  v-for="option of options"
  :key="option.value"
  :v-model="options"
  :value="option.value"
  :label="option.label" >
```

**Additional context**

We're using `:` so frequently it's an easy mistake to add it in front of an directive like `v-if` or `v-model`, the issue can be hard to notice. In case of the v-model the value just silently doesn't get updated.

**Comment by ota-meshi** [comment]
Thank you for your suggestions.

However, the rules that meet your needs are already implemented in 7.0.0-beta. So I will close this issue.

PR #1191
https://eslint.vuejs.org/rules/no-restricted-v-bind.html

**Comment by peteruithoven** [comment]
Awesome! Thanks for working on that merge request.
